src/Microsoft.Unity.Analyzers/NullableReferenceTypesSuppressor.cs

```diff
@@ -0,0 +1,166 @@
+
+//#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class NullableReferenceTypesSuppressor : DiagnosticSuppressor
+	{
+		private const string NULLABLE_WARNING = "CS8618";
+
+		internal static SuppressionDescriptor NullableRule { get; } = new SuppressionDescriptor("USP0016", NULLABLE_WARNING, Strings.NullableReferenceTypesSuppressorJustification);
+
+		//this, rather than IA => IA.Create(), should be faster because it doesn't allocate a new IA every time the value is needed
+		public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(NullableRule);
+
+		public override void ReportSuppressions(SuppressionAnalysisContext context)
+		{
+			foreach (Diagnostic diagnostic in context.ReportedDiagnostics.Where(diagnostic => diagnostic.Id == NULLABLE_WARNING))
+			{
+				SyntaxNode root = diagnostic.Location.SourceTree.GetRoot();
+				SyntaxNode location = root.FindNode(diagnostic.Location.SourceSpan);
+
+				//does the inspected class inherit from Monobehaviour?
+				if (!location.FirstAncestorOrSelf<ClassDeclarationSyntax>()?.BaseList?.ChildNodes().Any(node => GetSymbol(diagnostic.Location.SourceTree, ((BaseTypeSyntax)node).Type, context.Compilation).Extends(typeof(UnityEngine.MonoBehaviour))) ?? true)
+				{
+					continue;
+				}
+
+				PropertyDeclarationSyntax propertyDeclaration = location.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+
+				if (!(propertyDeclaration is null))
+				{
+					AnalyzeProperties(propertyDeclaration, diagnostic, context, root);
+					continue;
+				}
+
+				if (location is VariableDeclaratorSyntax fieldDeclaration)
+				{
+					AnalyzeFields(fieldDeclaration, diagnostic, context, root);
+					continue;
+				}
+
+				//TODO handle nullable warnings for constructors => diagnostic location is now on constructor
+				//TODO handle other Unity objects that cannot be initialized => e.g. Unity's new Input system
+				//TODO annotate nullability once this project actively uses C# 8.0
+			}
+		}
+
+		private static void AnalyzeFields(VariableDeclaratorSyntax declarator, Diagnostic diagnostic, SuppressionAnalysisContext context, SyntaxNode root)
+		{
+			FieldDeclarationSyntax declarationSyntax = declarator.FirstAncestorOrSelf<FieldDeclarationSyntax>();
+
+			//suppress for fields that are not private and not static => statics cannot be set in editor and are not shown in the inspector and cannot be set there
+			if (!declarationSyntax.Modifiers.Any(modifier => modifier.Text == "private" || modifier.Text == "static")
```

Comment:
You can use SyntaxKind enumeration instead of using strings: .Modifiers.Any(SyntaxKind.StaticKeyword)

Reply:
Thanks for the tip. I've already updated my code accordingly.